chess_gui.py

```diff
@@ -267,5 +238,40 @@ def draw_text(screen, text):
     screen.blit(text_object, text_location)
 
 
+def check_all_pieces(game_state, player, color):
+    count_b = 0
+    count_w = 0
+    for r in range(DIMENSION):
+        for c in range(DIMENSION):
```

Comment:
Rename 'count_b' and 'count_w' to 'count_black' and 'count_white' 
Try to avoid the usage of singe char variable names